async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -2181,9 +2181,14 @@ impl std::fmt::Display for BatchErrorKind {
 
 pub type BatchError = Error<BatchErrorKind>;
 
-impl From<SubscribeError> for BatchError {
-    fn from(err: SubscribeError) -> Self {
-        BatchError::with_source(BatchErrorKind::Subscribe, err)
+impl From<RequestError> for BatchError {
+    fn from(err: RequestError) -> Self {
+        match err.kind {
+            RequestErrorKind::SubscriptionFailed => {
+                BatchError::with_source(BatchErrorKind::Subscribe, err)
+            }
+            _ => panic!("Cannot convert error"),
```

Comment:
Here you can see that the irrelevant kinds of Error creates a possible panic scenario, or at least a path you need to handle in some way.